webapp/src/services/BaseService.ts

```diff
@@ -49,7 +50,7 @@ export class BaseService {
                 source.close()
```

Comment:
might want to check if this also removes the eventlistener, to avoid potential memory leaks